File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.java
class Solution {
    public int solution(int a, int b) {
        if(a * b % 2 == 0) {
            if(a % 2 == 0 && b%2 ==0) {
                return Math.abs(a - b);
            } else {
                return 2*(a+b);
            }
        }
        return a * a + b * b;
    }
}

File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.py
import math 

def solution(a, b):
    if a % 2 != 0 and b % 2 != 0:
        return math.pow(a, 2) + math.pow(b, 2)
    elif a % 2 == 0 and b % 2 == 0:
        return abs(a - b)
    else:
        return 2 * (a + b)